packages/teams-js/src/public/pages.ts

```diff
@@ -690,4 +690,72 @@ export namespace pages {
       return runtime.supports.pages ? (runtime.supports.pages.appButton ? true : false) : false;
     }
   }
+
+  /**
+   * Parameters for the NavigatewithinApp API
+   *
+   * @beta
+   */
+  export interface NavigateWithinAppParams {
+    /**
+     * @deprecated
+     * As of 2.0.0, please use {@link app.PageInfo.id | app.Context.page.id} instead
+     *
+     * Developer-defined ID of the Page to navigate to within the application (Formerly EntityID)
+     */
+    pageId: string;
+
+    /**
+     * @deprecated
+     * As of 2.0.0, please use {@link app.PageInfo.subPageId | app.Context.page.subPageId} instead
+     *
+     * Optional developer-defined ID describing the content to navigate to within the Page. This will be passed
+     * back to the application via the Context object.
+     */
+    subPageId?: string;
+  }
+
+  /**
+   * Provides functions for navigating without your own application ID.
+   *
+   * @beta
+   */
+  export namespace navigation {
```

Comment:
I like the name navigation, but I am worried that some of the other functions in pages (pages.navigateToApp, pages.tabs.navigateToTab) sound like they should also be in a namespace called navigate. Do you think this namespace name will end up being confusing for app developers?

Reply:
(Naming might be a good thing to check with Saurabh or PM about? Ultimately I would defer to them as long as they're aware of the potential conflict around the word "navigation")

Reply:
After careful consideration we are moving forward with 2 APIs `pages.navigate.to` and `pages.navigate.toDefaultPage`